### PR TITLE
Fixes #206 add support for deploying to heroku from travis

### LINKED ISSUE #206: Add support for deploying to heroku from travis
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

Currently deploying to heroku is limited to the webUI

### Expected behaviour

It should also be supported from travis for continuous deployment

### PR DESCRIPTION
**Pull request by pri22296**
## Description

Adds support for deploying to heroku from travis

Needs to have the following  environment variables set
- `HEROKU_APP_NAME`
- `HEROKU_API_KEY`

App name can also be specified with the `.yaydoc.yml` file by

```
publish:
  heroku:
     app_name: <app_name>
```

Also need to add the following in .travis.yml

```
sudo: required

before_deploy:
  - wget -qO- https://toolbelt.heroku.com/install-ubuntu.sh | sh
```

## Related Issue

Fixes #206 

## How Has This Been Tested?

Travis build log: - https://travis-ci.org/pri22296/testYaydoc2/builds/249101172
heroku app: - https://mytestyaydoc23.herokuapp.com/

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by pri22296** [comment]
@imujjwal96 @sch00lb0y Please review

**Comment by pri22296** [comment]
I did try https://github.com/travis-ci/apt-package-whitelist/issues/375#issuecomment-128055509 so that sudo could be avoided but the installation failed it.

**Comment by imujjwal96** [comment]
Update the README?

**Comment by pri22296** [comment]
@imujjwal96 updated

File: README.md [hunk 1]
```diff
@@ -80,6 +82,16 @@ deploy:
     script: wget -O - https://raw.githubusercontent.com/fossasia/yaydoc/master/generate_ci.sh | bash
 ```
 
+### Deploying to Heroku
+If you are deploying to Heroku, you also need to add the following content to the `.travis.yml` file
```
Comment (imujjwal96) [inline README.md]: you need to install `Heroku CLI`.........

Comment (pri22296) [inline README.md]: Do you want me to update the README?

**Comment by pri22296** [comment]
@sch00lb0y could you please review?

File: README.md (hunk 1 above)
Comment (imujjwal96) [inline README.md]: I think you should. It'll be more clear for those applying this change

Comment (pri22296) [inline README.md]: updated

**Pull request merged**